Tests/UnitTests/Purchasing/Purchases/BasePurchasesTests.swift

```diff
@@ -42,10 +42,15 @@ class BasePurchasesTests: TestCase {
         self.receiptFetcher = MockReceiptFetcher(requestFetcher: self.requestFetcher, systemInfo: systemInfoAttribution)
         self.attributionFetcher = MockAttributionFetcher(attributionFactory: MockAttributionTypeFactory(),
                                                          systemInfo: systemInfoAttribution)
-        self.backend = MockBackend(httpClient: MockHTTPClient(systemInfo: self.systemInfo,
-                                                              eTagManager: MockETagManager()),
-                                   apiKey: "mockAPIKey",
-                                   attributionFetcher: self.attributionFetcher)
+
+        let apiKey = "mockAPIKey"
+        let httpClient = MockHTTPClient(systemInfo: self.systemInfo, eTagManager: MockETagManager())
+        let config = BackendConfiguration(apiKey: apiKey,
```

Comment:
This is why I made the comment above regarding the api key. It seems weird that we need to pass it even though we're creating the `HTTPClient` already.

Reply:
It's weird because the HTTPClient doesn't get instantiated with an APIKey, so if you want auth-headers, you need to pass in an apikey that we can use with the `HTTPClient.authorizationHeader(withAPIKey apiKey: String) ` method. I'll refactor this to make more sense.